### PR TITLE
Move middleware to a sub package

### PR DESCRIPTION
**Pull request by neb42**
Move middleware to a sub package and rename files/functions to remove the "middleware" prefix/suffix

### CODE REVIEW

File: internal/server/middleware/auth.go [hunk 1]
```diff
@@ -48,7 +48,7 @@ type IDToken interface {
 	Claims(v interface{}) error
 }
 
-func authMiddleware(
+func AuthMiddleware(
```
Comment (ludydoo) [inline internal/server/middleware/auth.go:51]: could rename to `Authentication` (middleware suffix is redundant)

File: internal/server/middleware/countries.go [hunk 2]
```diff
@@ -9,7 +9,7 @@ import (
 	"go.uber.org/zap"
 )
 
-func countriesMiddleware(countryRepo db.CountryRepo) func(handler http.Handler) http.Handler {
+func CountriesMiddleware(countryRepo db.CountryRepo) func(handler http.Handler) http.Handler {
```
Comment (ludydoo) [inline internal/server/middleware/countries.go:12]: Could rename to `Countries` or perhaps `PrefetchCountries`

File: internal/server/middleware/logging.go [hunk 3]
```diff
@@ -1,10 +1,10 @@
-package server
+package middleware
 
 import (
 	"net/http"
 )
 
-func logMiddleware(h http.Handler) http.Handler {
+func LogMiddleware(h http.Handler) http.Handler {
```
Comment (ludydoo) [inline internal/server/middleware/logging.go:7]: Could rename to `Log` or `RequestLogging`

File: internal/server/middleware/permissions.go [hunk 4]
```diff
@@ -12,7 +12,7 @@ import (
 )
 
 // permissionMiddleware will compute the user's permissions and add them to the context
-func permissionMiddleware(
+func PermissionMiddleware(
```
Comment (ludydoo) [inline internal/server/middleware/permissions.go:15]: Could rename to `ComputePermissions`

File: internal/server/middleware/request_id.go [hunk 5]
```diff
@@ -7,7 +7,7 @@ import (
 	"github.com/nrc-no/notcore/internal/utils"
 )
 
-func requestIdMiddleware(h http.Handler) http.Handler {
+func RequestIdMiddleware(h http.Handler) http.Handler {
```
Comment (ludydoo) [inline internal/server/middleware/request_id.go:10]: Could rename to `RequestID`

File: internal/server/middleware/auth.go [hunk 6]
```diff
@@ -34,7 +34,7 @@ type idTokenVerifier struct {
 	verifier *oidc.IDTokenVerifier
 }
 
-func newIDTokenVerifier(verifier *oidc.IDTokenVerifier) IDTokenVerifier {
+func NewIDTokenVerifier(verifier *oidc.IDTokenVerifier) IDTokenVerifier {
```
Comment (ludydoo) [inline internal/server/middleware/auth.go:37]: I think the `IDTokenVerifier` and `IDToken` interfaces should still be in the `server` package

**Comment by neb42** [comment]
Thought I did remove all those `Middleware` suffixes, but I must have forgot

**Pull request merged**